models/user.go

```diff
@@ -520,7 +520,11 @@ func (u *User) UploadAvatar(data []byte) error {
 	}
 
 	u.UseCustomAvatar = true
-	u.Avatar = fmt.Sprintf("%x", md5.Sum(data))
+	// Different users can upload same image as avatar
+	// If we prefix it with u.ID, it will be separated
+	// Otherwise, if any of the users delete his avatar
+	// Other users will lose their avatars too.
+	u.Avatar = fmt.Sprintf("%d-%x", u.ID, md5.Sum(data))
```

Comment:
What's the point of using the md5 hash if the user won't be having more than one avatar? Can't this be just the user ID?

Reply:
For random avatars, user id is used as avatar name while for custom avatar, md5 hash is used for some security reasons. 
https://github.com/go-gitea/gitea/blob/1ae25c086e9e085cf44b71720211bd836300b373/models/user.go#L352-L356

Reply:
@masudur-rahman but doesn't your PR add the user ID to the file name nevertheless? If the goal is to hide the user ID perhaps it can be done like this:
```
u.Avatar = fmt.Sprintf("%x", md5.Sum(fmt.Sprintf("%d-%x", u.ID, md5.Sum(data))))
```
Or something around those lines.

Reply:
That seems reasonable. 
For migration, I checked if the avatar already has user ID prefixed with it. If not then did the migration.
But if I use md5 hash of the whole avatar name, I can't differentiate if it has user id prefixed to it. In fact as previously there was no user id prefixed with avatar name, I can just migrate all the avatar name. 

So, I am going to use the format you suggested.
```
u.Avatar = fmt.Sprintf("%x", md5.Sum(fmt.Sprintf("%d-%x", u.ID, md5.Sum(data))))
```

Reply:
@guillep2k let me know if it's fine.
Thanks...